### ISSUE TITLE
Error with start_url

### ISSUE THREAD

**Issue by aristath** [xwp/pwa-wp#115]
No matter what I do I keep getting this error on my lighthouse audits:
`Service worker does not successfully serve the manifest's start_url, No usable web app manifest found on page.`

Here's what I've managed to gather for now:

From https://developer.mozilla.org/en-US/docs/Web/Manifest#start_url:
> `start_url`: The URL that loads when a user launches the application (e.g. when added to home screen), typically the index. **Note that this has to be a relative URL, relative to the manifest url**.

Our manifest gets added on a URL like `https://domain.com/wp-json/wp/v2/web-app-manifest`, and `start_url` is `get_home_url()`

Should we not use a non-pretty URL (`https://domain.com/?rest_route=/wp/v2/web-app-manifest`) for the manifest, and `.` for the `start_url` ?

**Comment by westonruter** [comment]
I missed the relative URL requirement before.

On https://developers.google.com/web/fundamentals/web-app-manifest/#start-url I see this example:

```
"start_url": "/?utm_source=a2hs"
```

Could the manifest URL be left unchanged (e.g. as `https://domain.com/wp-json/wp/v2/web-app-manifest`) but then just strip the hostname off of the `start_url` (e.g. `/`)? (Note this would need to account for subdirectory installs as well, so we couldn't use `/` as a fixed string.)

**Comment by aristath** [comment]
Well, ideally the manifest would be on `https://domain.com/?get_webapp_manifest` or something similar, and `start_url` would be `.`.
Internally we can still use the rest-api response, but if the manifest's URL is `/wp-json/wp/v2/web-app-manifest` then I don't see how a relative URL to that path would make sense to browsers and be functional

**Comment by westonruter** [comment]
A URL like `/` is still relative in the required sense, I believe, as it is host-relative. And from the example on Google's docs, it is OK. Does it not work for you?

**Comment by aristath** [comment]
Interestingly enough, installing as an app works fine on my phone even using the absolute url that it has now. It's just that annoying notice in the audits that makes me question everything

**Comment by westonruter** [comment]
Does the audit complain about `/`?

**Comment by aristath** [comment]
Yeah, it still complains with `/` :thinking: 
For reference I was testing on a multisite so I changed `get_home_url()` to
```php
trailingslashit( wp_parse_url( get_home_url() )['path'] )
```

**Comment by westonruter** [comment]
I'm making some inquiries.

**Comment by westonruter** [comment]
I've opened https://github.com/GoogleChrome/lighthouse/issues/4261#issuecomment-459615004

**Comment by westonruter** [comment]
@aristath When I run the PWA audits in Lighthouse for my test environment at `https://wordpressdev.lndo.site/` I get the following results:

![image](https://user-images.githubusercontent.com/134745/52140487-d152c580-2607-11e9-986e-6a5052f73cfb.png)

So I'm not seeing:

> Service worker does not successfully serve the manifest's start_url, No usable web app manifest found on page.

My manifest looks like this:

```json
{
    "background_color": "#fff",
    "description": "Just another WordPress site",
    "dir": "ltr",
    "display": "minimal-ui",
    "icons": [],
    "lang": "en-US",
    "name": "Test7",
    "short_name": "Test7",
    "start_url": "https://wordpressdev.lndo.site",
    "theme_color": "#fff"
}

```

**Comment by westonruter** [comment]
I supplied the missing logos, added HTTPS redirection on my test site, and fixed the offline page handling for my child theme and now I get:

![image](https://user-images.githubusercontent.com/134745/52141571-014f9800-260b-11e9-88a9-09dd9697c145.png)

My manifest now reads:

```html
{
    "background_color": "#fff",
    "description": "Just another WordPress site",
    "dir": "ltr",
    "display": "minimal-ui",
    "icons": [
        {
            "sizes": "192x192",
            "src": "https://wordpressdev.lndo.site/content/uploads/2019/02/cropped-google-logo-192x192.png",
            "type": "image/png"
        },
        {
            "sizes": "512x512",
            "src": "https://wordpressdev.lndo.site/content/uploads/2019/02/cropped-google-logo.png",
            "type": "image/png"
        }
    ],
    "lang": "en-US",
    "name": "Test7",
    "short_name": "Test7",
    "start_url": "https://wordpressdev.lndo.site",
    "theme_color": "#fff"
}
```

My manifest is located at `https://wordpressdev.lndo.site/wp-json/wp/v2/web-app-manifest`. I do not get any warning regarding “ No usable web app manifest found on page.”

**Comment by westonruter** [comment]
The theme I'm using is https://github.com/westonruter/twentyseventeen-westonson

The quick HTTPS redirect I added to my `wp-config.php` is:

```php
if ( empty( $_SERVER['HTTPS'] ) && isset( $_SERVER['HTTP_HOST'] ) && isset( $_SERVER['REQUEST_URI'] ) && 'cli' !== php_sapi_name() ) {
	header( 'Location: https://' . $_SERVER['HTTP_HOST'] . $_SERVER['REQUEST_URI'] );
	exit;
}
```

**Comment by westonruter** [comment]
@aristath @ingereck I believe I found the problem. Please try the fix in #135.

In short, if the URL defined in the `start_url` does not get served by the service worker when browser is offline, then the manifest is considered unusable. Because the offline page was broken after the upgrade to Workbox 4beta1, this I believe is why the Lighthouse PWA audit was complaining about this. Please test and see if this is fixed for you now.

Also, make sure that the service worker serves an offline page when you are offline. For example, in my [child theme](https://github.com/westonruter/twentyseventeen-westonson) I see the following when I put the browser into offline mode:

> ![image](https://user-images.githubusercontent.com/134745/52315037-0f364d80-296a-11e9-983c-6d2df1265beb.png)

Note how the theme has an [`offline.php`](https://github.com/westonruter/twentyseventeen-westonson/blob/master/offline.php) defined.
